llama.cpp

```diff
@@ -1316,8 +1316,8 @@ static bool llama_kv_cache_find_slot(
 
     while (true) {
         if (cache.head + n_tokens > n_ctx) {
+            n_tested += cache.size - cache.head;
```

Comment:
This might be wrong, but I really don't understand the logic of setting `n_tested` to the full cache size in this case. That means we just give up searching in that case, but it's possible there's an empty slot of `n_tokens` at a point before the current `cache.head`, right?

At the least `n_tested += n_ctx - cache.head` is kind of redundant since `cache.head` just got set to 0. So this is like `n_tested += cache.head - 0`

Reply:
Oops, that was a bug! The fix is OK, but we should either use `cache.size` or `n_ctx` at all places in the function. No need to mix it.